sockeye/train.py

```diff
@@ -275,7 +275,8 @@ def create_data_iters_and_vocabs(args: argparse.Namespace,
             batch_num_devices=batch_num_devices,
             fill_up=args.fill_up)
 
-        check_condition(len(source_vocabs) == len(args.source_factors_num_embed) + 1,
+        check_condition(args.source_factors_combine == C.SOURCE_FACTORS_COMBINE_SUM \
```

Comment:
not sure about that logic. If combination method was concat, wouldn't this check always fail?

Reply:
If it's concat, then you have to specify the factor embedding sizes, so the next part of the check makes sure you have enough of them.